synapse/storage/state.py

```diff
@@ -60,6 +61,13 @@ def __init__(self, hs):
             self.STATE_GROUP_INDEX_UPDATE_NAME,
             self._background_index_state,
         )
+        self.register_background_index_update(
+            self.CURRENT_STATE_INDEX_UPDATE_NAME,
+            index_name="current_state_events_member_index",
+            table="current_state_events",
+            columns=["state_key"],
```

Comment:
No because we're looking up the room. The query we're optimising is essentially:

```sql
SELECT * FROM current_state_events WHERE type = 'm.room.member' AND state_key = ?
```

The `type` is in the where clause.